README.md

```diff
@@ -49,3 +73,11 @@ f(i1, i2) = ColorMixture{N0f8}((fluorophore_rgb"EGFP", fluorophore_rgb"tdTomato"
 Note the absence of `[]` brackets around the fluorophore names. For such constructors, `N0f8` is the only option if you're
 looking up the RGB values with `fluorophore_rgb`; however, if you hard-code the RGB values there is no restriction
 on the element type.
+
+## Why are the RGB colors encoded in the *type*? Why not a value field?
+
+In many places, JuliaImages assumes that you can convert from one color space to another purely from knowing the type you want to convert to. This would not be possible if the RGB colors were encoded as a second field of the color.
+
+## Why does this package require a minimum of Julia 1.7?
+
+To achieve good performance, the RGB values must be aggressively constant-propagated, a feature available only on Julia 1.7 and higher.
```

Comment:
The first time we're declaring incompatibility to LTS :) This almost means we're catching up with the latest features, but users can be unhappy :)

One trick I used in ImageIO to support <1.3 Julia versions is to register an empty placeholder package with only the type name, so that we can still add this package to, e.g., ImageCore, and thus we don't need the `@requires` here.

Then we can just safely declare that this is something that requires Julia >=1.7.

Just one small trick to work around the Pkg resolving mechanism and delay the incompatibility issue to runtime.

Reply:
It's not that so much. There is `Compat.@constprop` that allows one to run on 1.6. ~But here's what happens:~ oh wow, it's not what I thought! Originally the performance was dreadful on 1.6 (like, 100x slower), and to me that was so big that I thought we'd want to clearly indicate that the package is effectively broken on 1.6. But being more willing to generate errors appears to have fixed the performance. I don't fully understand why, but I'm happy for it.